src/state.rs

```diff
@@ -268,7 +277,7 @@ impl VMState {
             .map
             .iter()
             .filter_map(|(key, value)| {
-                let initial_value = self.storage_read_inner(key);
+                let initial_value = self.storage.borrow_mut().storage_read(key);
```

Comment:
This is outside the scope, so it's not necessary to change it here. However, I'll add a comment to link this to a future issue. The `storage_read()` operation should not require a mutable reference.

Reply:
Noted, I'll make sure to add a comment to ensure we correct this in a future refactor!